### DISCUSSION TITLE
mysqlclient (2.2.0) not supporting PEP 517 builds

### DISCUSSION THREAD

**Discussion by violuke** [PyMySQL/mysqlclient#623] [General]
### Describe the bug

Cannot install 2.2.0 with poetry in a debian docker container running on MacOS silicon.

### To Reproduce
```bash
poetry add mysqlclient
```

Gives error

```
Note: This error originates from the build backend, and is likely not a problem with poetry but with mysqlclient (2.2.0) not supporting PEP 517 builds. You can verify this by running 'pip wheel --use-pep517 "mysqlclient (==2.2.0)"'
```

---

I've seen the posts about installing `pkg-config`, but shouldn't I be able to install this via Poetry with that? And should PEP 517 be supported?

Thank you

**Comment by methane** [comment]
Even with poetry, you need pkg-config.
This package suports PEP 517. You just need properly configured pkg-config.

**Reply by methane** [reply]
Example of `pip wheel --use-pep517`

```
inada-n@PC-4524 ~> mkdir t
inada-n@PC-4524 ~> cd t
inada-n@PC-4524 ~/t> python3 -m venv --prompt . .venv
inada-n@PC-4524 ~/t> .venv/bin/pip wheel --use-pep517 mysqlclient
Collecting mysqlclient
  Downloading mysqlclient-2.2.0.tar.gz (89 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 89.5/89.5 kB 6.6 MB/s eta 0:00:00
  Installing build dependencies ... done
  Getting requirements to build wheel ... done
  Installing backend dependencies ... done
  Preparing metadata (pyproject.toml) ... done
Building wheels for collected packages: mysqlclient
  Building wheel for mysqlclient (pyproject.toml) ... done
  Created wheel for mysqlclient: filename=mysqlclient-2.2.0-cp311-cp311-macosx_13_0_arm64.whl size=75481 sha256=788c7900ff0fa98e269ec6f9c3962b3558b45756c0640976929788bd34432559
  Stored in directory: /Users/inada-n/Library/Caches/pip/wheels/a4/97/41/1e55ebbb29ce69d9008782f457658f48755b861fa145538ca0
Successfully built mysqlclient

[notice] A new release of pip is available: 23.0.1 -> 23.1.2
[notice] To update, run: python3.11 -m pip install --upgrade pip
```

**Reply by methane** [reply]
Example of `poetry add mysqlclient`:

```
inada-n@PC-4524 ~> poetry new t
Created package t in t
inada-n@PC-4524 ~> cd t
inada-n@PC-4524 ~/t> poetry add mysqlclient
Creating virtualenv t-z-NWPS82-py3.11 in /Users/inada-n/Library/Caches/pypoetry/virtualenvs
Using version ^2.2.0 for mysqlclient

Updating dependencies
Resolving dependencies... (0.1s)

Package operations: 1 install, 0 updates, 0 removals

  • Installing mysqlclient (2.2.0)

Writing lock file
```

**Reply by agarwalshashwat** [reply]
Hi @methane ! This still doesn't works for python3.12, current version we are trying to install is 2.2.4 .

**Comment by violuke** [comment]
Ah ok, should this not be a breaking change then? i.e. v3.0.0?

**Reply by methane** [reply]
![](https://imgs.xkcd.com/comics/workflow.png)

**Reply by methane** [reply]
"Every change breaks someone's workflow."

v3 is for major API/architecture redesign. Not for small workflow chagne.

Note that MySQL 8.0.33 breaked our build. If require pkg-config should be mysqlclient v3, MySQL 8.0.33 must be MySQL 41 (8 + 33).

**Comment by garciadias** [comment]
I have the same problem with a normal CPU. I don't think this is an Apple Silicon issue.
Sorry, I don't understand the comments. Is there a solution to this?

**Reply by garciadias** [reply]
I  had to run this to get it working:
`MYSQLCLIENT_CFLAGS="-I/usr/include/mysql" MYSQLCLIENT_LDFLAGS="-L/usr/lib/x86_64-linux-gnu -lmysqlclient" pip install mysqlclient`
